docs/iris/example_code/Meteorology/inset_plot.py

```diff
@@ -0,0 +1,73 @@
+"""
+Test Data Showing Inset Plots
+=============================
+
+This example demonstrates the use of a single 3D data cube with time, latitude
+and longitude dimensions to plot a latitude-averaged temperature series with
+an inset plot of the data region.
+
+"""
+
+###############################################################################
+
+import matplotlib.pyplot as plt
+import numpy as np
+import iris
+import cartopy.crs as ccrs
+import iris.quickplot as qplt
+import iris.plot as iplt
+
+###############################################################################
+
+
+def main():
+    iris.FUTURE.netcdf_promote = True
+    cube1 = iris.load_cube(iris.sample_data_path('ostia_monthly.nc'))
+    # Load the data
+    plot_cube1 = cube1[-1, :, :]
+    # Use only one time slice for data analysis, making a 2D cube
+    plot_line = plot_cube1.collapsed('latitude', iris.analysis.MEAN)
+    # Average over latitude to make a 1D cube
+    region = cube1[0, :, :]
+    # Separately, find region by using only one time point to make a 2D cube
+
+    fig = plt.figure()
+    # figure is the "window" you use.
+    ax_main = fig.add_subplot(1, 1, 1)
+    # add a single subplot (axes). Could also use "ax_main = plt.subplot()"
+    qplt.plot(plot_line)
+    # produce a quick plot of the 1D cube
+
+
+    xlist = (plot_line.coord('longitude')).points
+    # retrieve list of longitude coordinates to get maximum x-value for plot
+    ax_main.set_xlim(0, np.max(xlist))
+    ax_main.set_ylim(295, 310)
+    # adjust the y limits so that the inset map won't clash with main plot
+    ax_main.grid()
+    # add grid lines
+
+    # add a second set of axes specifying the fractional coordinates within
+    # the figure "fig" with bottom left corner at x=0.55, y=0.57 with width
+    # 0.3 and height 0.25.
+    # Also specify the projection
+    ax_sub = fig.add_axes([0.55, 0.57, 0.3, 0.25], projection=ccrs.Mollweide())
+
+    # Use iris.plot (iplt) here so we can adjust colour bar properties
+    iplt.pcolormesh(region, cmap='Blues')
+    # Use a sequential colour scheme to reduce confusion for those with
+    # colour-blindness
+    ticklist = np.linspace(np.min(region.data), np.max(region.data), 4)
+    plt.colorbar(orientation='horizontal', ticks=ticklist)
+    # manually set the orientation and tick marks on your colour bar
+    ax_sub.set_title('Region')
+    ax_sub.coastlines()
```

Comment:
On second thought this is a bad idea because your average includes the land points.